src/include/platform/NetworkCommissioning.h

```diff
@@ -231,8 +231,8 @@ class WirelessDriver : public Internal::BaseDriver
     virtual uint8_t GetScanNetworkTimeoutSeconds()    = 0;
     virtual uint8_t GetConnectNetworkTimeoutSeconds() = 0;
 
-    virtual Status RemoveNetwork(ByteSpan networkId)                 = 0;
-    virtual Status ReorderNetwork(ByteSpan networkId, uint8_t index) = 0;
+    virtual Status RemoveNetwork(ByteSpan networkId, MutableCharSpan outDebugText, uint8_t * outNetworkIndex) = 0;
```

Comment:
So the implementations are assuming that they can not set that last outparam (outNetworkIndex) and the caller will have a valid value as a result?

This is a really weird API, but I can understand wanting it to reduce churn for now... but then the behavior needs to be _very_ clearly documented, so someone doesn't call it with a pointer to un-initialized memory.

Same for the other functions that have this outNetworkIndex param.